freestyle-logging/shared/src/main/scala/logging.scala

```diff
@@ -21,21 +21,21 @@ object logging {
   @free
   trait LoggingM {
 
-    def debug(msg: String): FS[Unit]
+    def debug(msg: String, sourceAndLineInfo: Boolean = false)(implicit line: sourcecode.Line, file: sourcecode.File): FS[Unit]
```

Comment:
@suhasgaddam @ShaneDelmore We can support call by name semantics lifting args to `Function0` and that is probably the way to solve this. @suhasgaddam do you want to create an issue for that? thanks!